include/ppx/application.h

```diff
@@ -323,6 +323,33 @@ struct ApplicationSettings
             uint32_t     imageCount  = 2;
         } swapchain;
     } grfx;
+
+    // Default values for standard knobs
+    struct StandardKnobsDefaultValue
+    {
+        std::vector<std::string> assetsPaths     = {};
+        std::vector<std::string> configJsonPaths = {};
+        bool                     deterministic   = false;
+        bool                     enableMetrics   = false;
+        uint64_t                 frameCount      = UINT64_MAX;
+        int                      gpuIndex        = INT_MAX;
+#if !defined(PPX_LINUX_HEADLESS)
+        bool headless = false;
```

Comment:
There is already `headless` option above in `ApplicationSettings`, so it seems weird to have this twice. Which one takes priority if there are different, can they even be different.

Reply:
i feel in this case the knob one should be kept since this can be controlled both by cmdline and overwritten by the application. @angela28chen any thoughts on this?

Reply:
I think that the knob one should be kept. Also, I think there is a bug in the `benchmarks\headless_compute` example probably introduced when I changed up the commandline parsing a while ago to make use of knobs. I think the line `settings.headless = true;` in there is ignored, but after this CL it can just be fixed to `settings.standardKnobsDefaultValue.headless = true;`

Reply:
I feel that headless change would be a bit more involved, especially with the discussion by @hysw. I could do another PR to handle the headless, what do you think @apazylbe ?

Reply:
sgtm